project.php

```diff
@@ -45,7 +46,11 @@
 $DEFAULT_DETAIL_LEVEL = 3;
 
 // Validate all the input
-$projectid = get_projectID_param($_GET, 'id');
+try {
+    $projectid = get_projectID_param($_GET, 'id');
```

Comment:
When I was testing this by manipulating the url on a proofing page in page_temp state, if you put in an invalid projectid, then if the projectid was able to get as far as get_indicated_LPage an exception would be thrown and caught. But the invalid projectid is caught before that so it seemed to make sense to display the same abort page in that case.

Reply:
I appreciate trying to improve the user's error experience but I don't think we want these changes.

The only way an invalid (or not yet created) projectid can be passed in is if the user is manipulating the URL, in which case they are SOL and I would rather barf loudly with an error than try to do so gracefully. Using `abort()` here is perhaps worse since it will show links to a project page that will also be invalid (since the projectid is invalid) and the projectid output in the page isn't escaped since it is presumed to be invalid.

And a reminder that on PROD we never show the exception stack-trace, just the thrown message. eg: https://www.pgdp.net/c/tools/proofers/proof.php?projectid=invalid&proj_state=P1.proj_avail

Reply:
> I appreciate trying to improve the user's error experience but I don't think we want these changes.
> 
> The only way an invalid (or not yet created) projectid can be passed in is if the user is manipulating the URL, in which case they are SOL and I would rather barf loudly with an error than try to do so gracefully. Using `abort()` here is perhaps worse since it will show links to a project page that will also be invalid (since the projectid is invalid) and the projectid output in the page isn't escaped since it is presumed to be invalid.
> 
> And a reminder that on PROD we never show the exception stack-trace, just the thrown message. eg: https://www.pgdp.net/c/tools/proofers/proof.php?projectid=invalid&proj_state=P1.proj_avail

Good. undone changes to proof.php.